build.sbt

```diff
@@ -26,7 +26,6 @@ ThisBuild / scmInfo := {
   Some(ScmInfo(url(s"https://github.com/$slug"), s"git@github.com:$slug.git"))
 }
 ThisBuild / licenses := List(("Apache-2.0", url("https://www.apache.org/licenses/LICENSE-2.0")))
-ThisBuild / scalafmtOnCompile := true
```

Comment:
The reason why we turn this setting on is because when someone sends a PR, half the time they won't remember to run scalafmt. Maybe Scalafmt has been popularized enough that this is no longer an issue? I wish we could just append a format commit, which is what we do at work.

Reply:
> The reason why we turn this setting on is because when someone sends a PR, half the time they won't remember to run scalafmt. Maybe Scalafmt has been popularized enough that this is no longer an issue? I wish we could just append a format commit, which is what we do at work.

On the flip side, if someone does code changes and doesn't run compile then they can make a PR with unformatted code which can get merged (this actually happened with librarymanagement).

The intention here is that even if someone submits an unformatted PR, the PR won't actually be able to be merged (on the assumption you add it as a ci check).

Over the years I have tried many different approaches and find this one to be the most suitable for everyone's work style and actually enforced that the code base stays formatted which is the whole point.

Reply:
We run `scalafmtCheckAll` on CI, so I'm not sure how unformatted code got slipped? Maybe we don't enforce linear history so scalafmt version change didn't catch some PRs?

Reply:
I don't know how it happened but when we were doing the sbt 2.x breaking changes the codebase was not in a formatted state (which was quite annoying when working on it)

In any case, the setup this is suggested in this PR is the same as many sbt plugins and other OS projects as well (I have contributed this setup to many projects) so it shouldn't be too controversial.